src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -309,7 +308,31 @@ public void addMapComparison(ColumnMetadata def, ByteBuffer key, Operator op, By
 
         public void addGeoDistanceExpression(ColumnMetadata def, ByteBuffer point, Operator op, ByteBuffer distance)
         {
-            add(new GeoDistanceExpression(def, point, op, distance));
+            var primaryGeoSearch = new GeoDistanceExpression(def, point, op, distance);
+            // The following logic optionally adds a second search expression in the event that the query area
+            // crosses then antimeridian.
+            if (primaryGeoSearch.crossesAntimeridian())
+            {
+                var shiftedGeoSearch = primaryGeoSearch.buildShiftedExpression();
+                if (current.isDisjunction)
```

Comment:
why do we check isDisjunction for the crosses cases, but not the standard case?

Reply:
Because we're only adding an extra expression in the crosses case, and if it is a disjunction, we don't need to nest the primary and secondary searches.